staging/kos/pkg/controllers/connectionagent/network_interface.go

```diff
@@ -199,15 +173,20 @@ func (ifc *remoteNetworkInterface) String() string {
 }
 
 func (ca *ConnectionAgent) createLocalNetworkInterface(att *netv1a1.NetworkAttachment) (ifc *localNetworkInterface, statusErrs sliceOfString, err error) {
-	ifc = ca.newLocalNetworkInterfaceForAttachment(att)
+	ifc = &localNetworkInterface{}
+	ifc.VNI = att.Status.AddressVNI
+	ifc.GuestIP = gonet.ParseIP(att.Status.IPv4)
+	ifc.GuestMAC = generateMACAddr(ifc.VNI, ifc.GuestIP)
+	ifc.Name = generateIfcName(ifc.GuestMAC)
+	ifc.postDeleteExec = att.Spec.PostDeleteExec
```

Comment:
Why do this series of imperative statements rather than declare the data contents in the constructor?

Reply:
The value of certain fields (`ifc.GuestMAC` and `ifc.Name`) depends on the value of previously set fields, so you CANNOT set them directly in the constructor. Even fields that could be set in the constructor are set with an imperative statement for consistency.